File: Functional Programming/Presentasi/dekontruksi.py
angka = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

def rekursiveFilter(arry, genap, a=0) :
  if a == len(arry)-1:
    genap += [arry[a]]
    return genap
  else :
    if arry[a] % 2 == 0 :
      genap += [arry[a]]
    return rekursiveFilter(arry, genap, a+1)

hasil = rekursiveFilter(angka, [])
print(hasil)

File: Functional Programming/Presentasi/dekontruksi.dart
List<int> rekursiveFilter(List<int> arry, List<int> genap, int a){
  if(a == arry.length - 1){
    genap += [arry[a]];
    return genap;
  }
  else{
    if(arry[a] % 2 == 0){
      genap += [arry[a]];
    }
  }
  return rekursiveFilter(arry, genap, a+1);
}

void main(){
  List<int> angka = [1,4,7,2,5,8,3,6,9,10];
  List<int> hasil = rekursiveFilter(angka, [], 0);
  print(hasil);
}